lib/plugins/aws/deployFunction.js

```diff
@@ -103,10 +103,27 @@ class AwsDeployFunction {
       FunctionName: functionObj.name,
     };
 
-    if ('awsKmsKeyArn' in functionObj && !_.isObject(functionObj.awsKmsKeyArn)) {
-      params.KMSKeyArn = functionObj.awsKmsKeyArn;
-    } else if (serviceObj && 'awsKmsKeyArn' in serviceObj && !_.isObject(serviceObj.awsKmsKeyArn)) {
-      params.KMSKeyArn = serviceObj.awsKmsKeyArn;
+    let kmsKeyArn = functionObj.kmsKeyArn || providerObj.kmsKeyArn;
+    if (functionObj.awsKmsKeyArn) {
+      kmsKeyArn = !kmsKeyArn ? functionObj.awsKmsKeyArn : kmsKeyArn;
+
+      this.serverless._logDeprecation(
+        'FUNCTION_DEPRECATED_PROPERTY_AWS_KMS_KEY_ARN',
+        'Starting with next major version, awsKmsKeyArn function property will be replaced by function[].kmsKeyArn.'
+      );
+    }
+
+    if (serviceObj.awsKmsKeyArn) {
+      kmsKeyArn = !kmsKeyArn ? serviceObj.awsKmsKeyArn : kmsKeyArn;
+
+      this.serverless._logDeprecation(
+        'SERVICE_DEPRECATED_PROPERTY_AWS_KMS_KEY_ARN',
```

Comment:
Here we should use `AWS_KMS_KEY_ARN` code so it constructs a proper link to deprecation documentation